staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -331,9 +363,17 @@ loop:
 			if event.Type == watch.Error {
 				return apierrs.FromObject(event.Object)
 			}
-			if e, a := r.expectedType, reflect.TypeOf(event.Object); e != nil && e != a {
-				utilruntime.HandleError(fmt.Errorf("%s: expected type %v, but watch event object had type %v", r.name, e, a))
-				continue
+			if r.expectedType != nil {
+				if e, a := r.expectedType, reflect.TypeOf(event.Object); e != a {
+					utilruntime.HandleError(fmt.Errorf("%s: expected type %v, but watch event object had type %v", r.name, e, a))
+					continue
+				}
+			}
+			if r.expectedGVK != nil {
+				if e, a := r.expectedGVK, event.Object.GetObjectKind().GroupVersionKind(); e.String() != a.String() {
```

Comment:
sorry, just noticed this... is there a reason not to do this, rather than stringifying them?
```
if e, a := *r.expectedGVK, event.Object.GetObjectKind().GroupVersionKind(); e != a {
```

Reply:
Thank you for catching my elementary mistake.  :)